moveit_ros/planning/moveit_cpp/src/planning_component.cpp

```diff
@@ -189,7 +216,127 @@ PlanningComponent::PlanSolution PlanningComponent::plan(const PlanRequestParamet
   //    visual_tools_->publishRobotState(last_solution_trajectory_->getLastWayPoint(), rviz_visual_tools::TRANSLUCENT);
   //  }
   //}
-  return *last_plan_solution_;
+
+  if (update_last_solution)
+  {
+    last_plan_solution_ = plan_solution;
+  }
+  return *plan_solution;
+}
+
+PlanningComponent::PlanSolution PlanningComponent::plan(const MultiPipelinePlanRequestParameters& parameters,
+                                                        SolutionCallbackFunction solution_selection_callback,
+                                                        StoppingCriterionFunction stopping_criterion_callback)
+{
+  // Create solutions container
+  PlanningComponent::PlanSolutions planning_solutions{ parameters.multi_plan_request_parameters.size() };
+  std::vector<std::thread> planning_threads;
+  planning_threads.reserve(parameters.multi_plan_request_parameters.size());
+
+  // Launch planning threads
+  for (auto const& plan_request_parameter : parameters.multi_plan_request_parameters)
+  {
+    auto planning_thread = std::thread([&]() {
+      try
+      {
+        auto solution = plan(plan_request_parameter, false);
+        planning_solutions.pushBack(solution);
+      }
+      catch (const std::exception& e)
+      {
+        RCLCPP_ERROR_STREAM(LOGGER, "Planning pipeline '" << plan_request_parameter.planning_pipeline.c_str()
+                                                          << "' threw exception '" << e.what() << "'");
+        auto plan_solution = std::make_shared<PlanSolution>();
```

Comment:
This temporary plan_solution is not being added to the solutions list and doesn't seem to be used anywhere else

Reply:
Thanks!